### DISCUSSION TITLE
History and Metrics for Snapmirror Relationships

### DISCUSSION THREAD

**Discussion by KlausHub** [NetApp/harvest#1402] [Q&A] [answered]
Hi again,
i am running on 22.08 and tried to get some more info on specific (XDP-Volume) Snapmirror Relations.
![image](https://user-images.githubusercontent.com/97510878/199542536-14d0f038-72c8-4820-a213-bd7e0dc5087f.png)

After clicking on one off  the Relationsships/Volumes, i get a not found error/info.
![image](https://user-images.githubusercontent.com/97510878/199541857-d5bbd137-12a7-4924-b31b-c876a1e4da89.png)

many thanks in advance for your Feedback and Help on that.

Best Regards,
Klaus

**Comment by cgrinds** [comment]
Thanks for reporting @KlausHub.
I see what happened and we'll fix.

The way Grafana data links work is somewhat brittle. The URL include the dashboard's `uuid` which often changes when the dashboard does.

If you want to fix your local copy while we work on a permanent fix, here's what to do.
1. In the browser's address bar copy the dashboard's UUID - the part highlighted here 
<img width="812" alt="image" src="https://user-images.githubusercontent.com/242252/199560752-6c0f76d1-6d31-46ac-922d-5cc48c8f0882.png">
2. Move over the table and press `e` to edit the panel
3. Select Overrides in the top right and scroll down until you see the override for `ID`, it is `Override 17` for my dashboard
<img width="476" alt="image" src="https://user-images.githubusercontent.com/242252/199561219-896e94de-2fa2-4fee-9b73-08c651e048e8.png">
4. Press the pencil icon to open the `Edit Link` dialog and change the `URL` to match what you copied out of the browser's address bar, like this
<img width="866" alt="image" src="https://user-images.githubusercontent.com/242252/199561754-fa705d2b-508b-48cb-ad63-3c09dad24975.png">

5. Press `Save` and the links should work

**Comment by KlausHub** [comment]
Good Morning Chris,

thanks for the fix and prefix/guide how to do it by myself.
Now the link does work, but the page i get looks nearly the same.

I cannot find any specific info only to the relationship i clicked on.
I expected to get some info/graphs about Transferstatus (to see when it
started/stopped, ....) for this specific relationship.
[image: image.png]

Do i missed something or is there no dashboard for details of a single
Snapmirror Relationship (type: XDP, Ontap 9.8 or higher).

Best Regards,
Klaus

Am Mi., 2. Nov. 2022 um 18:38 Uhr schrieb Chris Grindstaff <
***@***.***>:

**Reply by cgrinds** [reply]
Ah! I see the confusion. Yes, clicking that link is the same as changing a variable at the top of the dashboard. In other words, it's a normal filter on the dashboard, in this case, clicking one of the `ID` links will open a new tab and prefill the `SourceCluster` variable with that row's source cluster.

**Reply by cgrinds** [reply]
Hi @KlausHub we've added Source and Destination cluster to the included table and removed ID.

Can you say more about what you'd like to see in terms of a dashboard for a single relationship?
That's something we can put on the roadmap for next release.

<img width="1028" alt="image" src="https://user-images.githubusercontent.com/242252/200840042-2b7eb614-e5e8-44a0-b90a-b8778bc9ee03.png">

**Comment by KlausHub** [comment, marked as answer]
HI Chris,

thanks, thats good news to have it more intuitive on first sight.

Regarding my wishlist for Details for a single Snapmirror Relationship i
would be interested in at least 3 Graphs
1.) Graph for Transfer status (to see the duration the Snapmirror was
transferring data)
2.) Graph for LAG-Time
3.) Graph for Transfered Data (total progress of currently transferred data
for this update or initial sync)
easy to identify situations with more than normal load

Best Regards,
Klaus

Am Mi., 9. Nov. 2022 um 14:15 Uhr schrieb Chris Grindstaff <
***@***.***>:

**Reply by rahulguptajss** [reply]
Thanks @KlausHub . I have opned a feature request for the same. #1461
